### ISSUE TITLE
Update to Unicode 13.

### ISSUE THREAD

**Issue by crlf0710** [rust-lang/rust#70002]
Now that Unicode 13 is released, maybe we should bump the unicode version.

**Comment by jonas-schievink** [comment]
https://github.com/rust-lang/rust/pull/69929

**Comment by est31** [comment]
@jonas-schievink That PR is missing https://github.com/rust-lang/rust/blob/9e6fb538f9254884ca9f958fdce413d6c3f2016c/src/librustc_parse/lexer/unicode_chars.rs

newest list is https://www.unicode.org/Public/security/13.0.0/confusables.txt

See also  #51080

**Comment by Mark-Simulacrum** [comment]
cc @estebank @rust-lang/wg-diagnostics re:confusables list

**Comment by est31** [comment]
It's a bit tough because that list isn't automatically generated, or it has been at a point in the past but since then it has been manually edited and changed. Those amendments are valuable and should not be lost by an automatic regeneration. One could think about having two lists though, one automatically generated, the other with the rust-specific edits/improvements.

**Comment by crlf0710** [comment]
As i've recently got a crate that has `confusables.txt` data in it. I made a small analysis on existing items in fore-mentioned list.  There're 178 items that are already covered by `confusables.txt`, 85 items not covered, and one duplicate item (Canadian Syllabics Final Middle Dot).

**Comment by crlf0710** [comment]
The former two lists [are here](https://github.com/crlf0710/unicode-security/blob/c697f2c9043504e613b958d29ad10aa89dc54d4e/src/tests.rs#L73-L422). Also cc @Manishearth to see whether the second list is of some value to further development of UTS39.

**Comment by Manishearth** [comment]
The in-tree list is specifically about things which are confusable with _rust_ syntax

**Comment by crlf0710** [comment]
Yes, and rust syntax here means most of ASCII punctuation characters. 

I saw one major difference between this list and the UTS39 one is that most of the items here are just unintentional, and not for security reasons. For example,
```text
('，', "Fullwidth Comma", ','),
```
I'm using an IME daily, and a single shift key press toggles between Latin mode and Han mode, where the comma key corresponds to these two characters in each mode. And it's really easy to unintentionally get them wrong. So while this item is not in UTS39 list, it is practically quite useful.

**Comment by crlf0710** [comment]
Updated the issue title to reflect the actual discussion topic here - the libs part is already done in #69929.
